Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp

```diff
@@ -173,6 +174,34 @@ URL ApplicationManifestParser::parseStartURL(const JSON::Object& manifest, const
     return startURL;
 }
 
+ApplicationManifest::Direction ApplicationManifestParser::parseDir(const JSON::Object& manifest)
+{
+    using enum ApplicationManifest::Direction;
+
+    auto value = manifest.getValue("dir"_s);
+    if (!value)
+        return Auto;
+
+    auto stringValue = value->asString();
+    if (!stringValue) {
+        logManifestPropertyNotAString("dir"_s);
+        return Auto;
+    }
+
+    static constexpr std::pair<ComparableLettersLiteral, ApplicationManifest::Direction> directionMappings[] = {
+        { "auto", Auto },
+        { "ltr", LTR },
+        { "rtl", RTL },
+    };
+    static constexpr SortedArrayMap directions { directionMappings };
+
+    if (auto* dirValue = directions.tryGet(StringView(stringValue).trim(isASCIIWhitespace<UChar>)))
+        return *dirValue;
+
+    logDeveloperWarning(makeString("\""_s, stringValue, "\" is not a valid dir."_s));
```

Comment:
It would be more efficient to use `'"'` instead of `"\""_s`, which would otherwise accomplish the same thing; some day maybe we can figure out how to get the compiler to warn about this or perhaps compile it into the same code to eliminate the efficiency difference.